src/query/fuzzy_query.rs

```diff
@@ -132,14 +132,44 @@ impl FuzzyTermQuery {
             });
 
         let term_value = self.term.value();
-        let term_text = term_value.as_str().ok_or_else(|| {
-            InvalidArgument("The fuzzy term query requires a string term.".to_string())
-        })?;
+        let term_value_string = term_value.as_str().is_some();
+        let term_value_json = term_value.as_json().is_some();
+        let term_text: String;
+        if term_value_string {
```

Comment:
this was a pretty hacky way to do it, but you can pretty easily separate the actual query term we're searching for from the query path. However it seems like the Automaton is not quite matching the term with any of the documents. I've found this part difficult to debug / verify so far

Reply:
I'm a little confused by the difference between `What’s more puzzling is how the JSON path is traversed` and this comment. 
- Are we able to traverse the path to get the term? or is this just working for a specific case and we need to extend it to handle other cases?

Reply:
i can't seem to figure out how it's traversed even in the TermQuery case - so basically I'm not sure if it's being traversed properly in the Automaton and the term is somehow incorrect (probably not) or if it's not being traversed in the Automaton at all